coderd/notifications/dispatch/webhook_test.go

```diff
@@ -0,0 +1,144 @@
+package dispatch_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+	"github.com/coder/serpent"
+
+	"github.com/coder/coder/v2/coderd/notifications/dispatch"
+	"github.com/coder/coder/v2/coderd/notifications/types"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestWebhook(t *testing.T) {
+	t.Parallel()
+
+	const (
+		titleTemplate = "this is the title ({{.Labels.foo}})"
+		bodyTemplate  = "this is the body ({{.Labels.baz}})"
+	)
+
+	msgPayload := types.MessagePayload{
+		Version:          "1.0",
+		NotificationName: "test",
+		Labels: map[string]string{
+			"foo": "bar",
+			"baz": "quux",
+		},
+	}
+
+	tests := []struct {
+		name          string
+		serverURL     string
+		serverTimeout time.Duration
+		serverFn      func(uuid.UUID, http.ResponseWriter, *http.Request)
+
+		expectSuccess   bool
+		expectRetryable bool
+		expectErr       string
+	}{
+		{
+			name: "successful",
+			serverFn: func(msgID uuid.UUID, w http.ResponseWriter, r *http.Request) {
+				var payload dispatch.WebhookPayload
+				err := json.NewDecoder(r.Body).Decode(&payload)
+				require.NoError(t, err)
```

Comment:
Since this is running in a goroutine, let's change these to `assert`.

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test.

Reply:
Oh you're right! The `http.Server` serves in a goroutine, whoops.